### ISSUE TITLE
[chore] Replace `Mutex` with `RwLock` for themes

### ISSUE THREAD

**Issue by cylewitruk** [fadeevab/cliclack#41]
Since the theme is write-once, read-many, I think a `RwLock` would be more appropriate than a `Mutex`. I remember running into deadlocks on accident when writing the multiprogress suggestion PR, and this would simply make it easier to avoid such caveats.

**Comment by fadeevab** [comment]
@cylewitruk It sounds reasonable, I remember running into deadlocks too, however, due to urgency, I evaded deadlocks via timely unlocking.

**Comment by cylewitruk** [comment]
> @cylewitruk It sounds reasonable, I remember running into deadlocks too, however, due to urgency, I evaded deadlocks via timely unlocking.

Yeah no I get it, I just ran into this with my own theme handling using Mutex and resolved with RwLock, hence I I bring it up here 😊
